### DISCUSSION TITLE
Question: `RESULTS_BACKEND` and `CELERY_RESULT_BACKEND` seem repeated? they look like the same thing but why there are two of them?

### DISCUSSION THREAD

**Discussion by 1c7** [apache/superset#18674] [Q&A / Help] [answered]
# CELERY_RESULT_BACKEND
<img width="1305" alt="image" src="https://user-images.githubusercontent.com/1804755/153367114-1881ebde-423e-4926-a95a-dd8b42af2859.png">

# RESULTS_BACKEND
<img width="1304" alt="image" src="https://user-images.githubusercontent.com/1804755/153367179-6bc9b166-7469-4fbf-8bc2-98fc8be3a82f.png">

# Question: Why there are `RESULTS_BACKEND` and `CELERY_RESULT_BACKEND`? in Document [Async Queries via Celery](https://superset.apache.org/docs/installation/async-queries-celery) They Look like the same thing

**Comment by 1c7** [comment]
# Celery Documentation
https://docs.celeryproject.org/en/stable/userguide/configuration.html#task-result-backend-settings   

Section Introduction
<img width="1238" alt="image" src="https://user-images.githubusercontent.com/1804755/153367728-1029ae5a-85c1-4e52-ba2c-5d26873d5022.png">

<img width="1193" alt="image" src="https://user-images.githubusercontent.com/1804755/153367769-5da7f269-14e2-4f68-822a-e55f69dd7156.png">

<img width="1207" alt="image" src="https://user-images.githubusercontent.com/1804755/153367854-28f2d6fc-dd08-40a4-9bdf-94503b20ad6c.png">

# Conclusion: Celery does support `CELERY_RESULT_BACKEND`
Because Celery Docs says:  

> Celery will still be able to read old configuration files until Celery 6.0.

<img width="861" alt="image" src="https://user-images.githubusercontent.com/1804755/153370065-959bb65f-d09c-40ed-b0e1-de04ad71ab49.png">


Right now Superset are using Celery `5.2.3`
So `CELERY_RESULT_BACKEND` still works in Celery 5.2.3

**Comment by 1c7** [comment]
## By tracking Superset sources code

## File `superset/extensions.py` are using config `app.config["RESULTS_BACKEND"]`
<img width="1440" alt="image" src="https://user-images.githubusercontent.com/1804755/153368460-b6cd12a5-8107-4871-b508-61aac4273ecf.png">

## `results_backend_manager` can be imported by other
<img width="1440" alt="image" src="https://user-images.githubusercontent.com/1804755/153368678-94b12e2a-e45d-481d-a0da-8c6d2228636d.png">

## Not sure what this does `results_backend_manager.init_app(self.superset_app)`
<img width="1440" alt="image" src="https://user-images.githubusercontent.com/1804755/153369254-a92daa2d-7f7f-452f-acd8-489d471d57d9.png">

# Conclusion: no idea where `RESULTS_BACKEND` would be effective.
because I don't know how to keep tracking after `results_backend_manager.init_app(self.superset_app)`

**Comment by villebro** [comment, marked as answer]
Those are different things. `CELERY_RESULT_BACKEND` is a Celery config option (notice it's a property in the `CeleryConfig` class) that specifies where celery should store its task results. RESULTS_BACKEND defines where asynchronously executed SQL Lab query results will be stored.

**Reply by shauryagoel** [reply]
Is there a way to change TTL of keys in redis of RESULTS_BACKEND?
